File: packages/pigeon/platform_tests/test_plugin/windows/test/non_null_fields_test.cpp
// Copyright 2013 The Flutter Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include <gtest/gtest.h>

#include "pigeon/non_null_fields.gen.h"

namespace non_null_fields_pigeontest {

TEST(NonNullFields, Build) {
  NonNullFieldSearchRequest request("hello");

  EXPECT_EQ(request.query(), "hello");
}

TEST(NonNullFields, Equality) {
  NonNullFieldSearchRequest request1("hello");
  NonNullFieldSearchRequest request2("hello");
  NonNullFieldSearchRequest request3("world");

  EXPECT_EQ(request1, request2);
  EXPECT_NE(request1, request3);
}

}  // namespace non_null_fields_pigeontest

File: packages/pigeon/platform_tests/shared_test_plugin_code/test/non_null_fields_test.dart
// Copyright 2013 The Flutter Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'package:flutter_test/flutter_test.dart';
import 'package:shared_test_plugin_code/src/generated/non_null_fields.gen.dart';

void main() {
  test('test constructor', () {
    final request = NonNullFieldSearchRequest(query: 'what?');
    expect(request.query, 'what?');
  });

  test('test equality', () {
    final request1 = NonNullFieldSearchRequest(query: 'hello');
    final request2 = NonNullFieldSearchRequest(query: 'hello');
    final request3 = NonNullFieldSearchRequest(query: 'world');

    expect(request1, request2);
    expect(request1, isNot(request3));
    expect(request1.hashCode, request2.hashCode);
  });
}
